action.yml

```diff
@@ -88,22 +88,32 @@ runs:
       env:
         IS_NON_COMMIT_ARG: ${{ fromJSON(fromJSON(inputs.client_payload)).isNonCommitEvent }}
         ENABLE_CACHE_ARG: ${{ env.ENABLE_CACHE }}
+        RUN_ID_ARG: ${{ fromJSON(fromJSON(inputs.client_payload)).artifactRunId }}
       with:
         script: |
           require('${{ github.action_path }}/scripts/get-condition-vars.js')(core);
 
     - name: Download cache artifact
+      id: download-cache
       uses: actions/download-artifact@v4
       if: ${{ env.SKIP_GIT_CLONE == 'true' }}
+      continue-on-error: true
       with:
         github-token: ${{ fromJSON(fromJSON(inputs.client_payload)).githubToken || github.token }}
         repository: ${{ inputs.full_repository }}
         run-id: ${{ fromJSON(fromJSON(inputs.client_payload)).artifactRunId }}
         name: output
         path: code/output
 
+    - name: Check if download cache artifact failed
+      uses: actions/github-script@v7
+      env:
+        ARTIFACT_OUTCOME_ARG: ${{ steps.download-cache.outcome }}
+      with:
+        script: require('${{ github.action_path }}/scripts/check-cache-download-status')(core);
+
     - name: Checkout Pull Request branches history
-      if: ${{ env.SKIP_GIT_CLONE == 'false' }}
+      if: ${{ env.SKIP_GIT_CLONE == 'false' || env.CACHE_DOWNLOAD_FAILED == 'true' }}
```

Comment:
this condition `env.SKIP_GIT_CLONE == 'false' || env.CACHE_DOWNLOAD_FAILED == 'true'` repeat itself 3 times, I think `env.CACHE_DOWNLOAD_FAILED == 'true'` maybe be in the logic when you calculated `SKIP_GIT_CLONE`

Reply:
but I have to calculate `SKIP_GIT_CLONE` before downloading artifact... I can do other step that calculate it